lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -2542,6 +2542,17 @@ void IRGenDebugInfoImpl::emitDbgIntrinsic(
       !llvm::FindDbgDeclareUses(Storage).empty())
     return;
 
+  // Fragment DIExpression cannot cover the whole variable
+  // or going out-of-bound.
+  if (auto Fragment = Expr->getFragmentInfo())
```

Comment:
Do we understand why we generate a fragment that covers the entire variable? Does this happen because we SROA a struct with a single member? Could we catch that case in `IRGenDebugInfoImpl::emitVariableDeclaration()` and not emit a DW_OP_LLVM_fragment?

Reply:
Yes this only happens when the struct only has one member. Originally I was going to solve this by preserving the bail out logic here and avoid emitting DW_OP_LLVM_fragment some place earlier as suggested. But then I found that it will be easier to just drop the fragment part here.

Reply:
That works!